server/src/endpoints/qs/ws/mod.rs

```diff
@@ -225,7 +229,7 @@ pub(crate) async fn upgrade_connection(
     match ws::start(qs_ws_connection, &req, stream) {
         Ok(res) => res,
         Err(e) => {
-            tracing::error!("Error upgrading connection: {}", e);
+            error!("Error upgrading connection: {}", e);
```

Comment:
If you're touching these lines anyway, you might as well turn it into structured logging. Same for the two above.